plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/model/derivedfeatures/ScenarioModelDerivedFeatures.java

```diff
@@ -13,16 +13,20 @@
 import hu.bme.mit.gamma.scenario.model.Signal;
 
 public class ScenarioModelDerivedFeatures extends ExpressionModelDerivedFeatures {
-
-	// TODO use javaUtil.filterInto list instead of these casts
 	
-	public static InteractionDirection getDirection(ModalInteractionSet set) {
+	public static final ScenarioModelDerivedFeatures INSTANCE = new ScenarioModelDerivedFeatures();
```

Comment:
`..DerivedFeatures` classes are generally static classes in Gamma, e.g., `ExpressionModelDerivedFeatures`. See if you could refactor this class that way.